enumeratum-json4s/src/main/scala/enumeratum/values/Json4s.scala

```diff
@@ -0,0 +1,67 @@
+package enumeratum.values
+
+import org.json4s.CustomSerializer
+import org.json4s.JsonAST.{JInt, JLong, JString}
+
+
+object Json4s {
+
+  def serializer[A <: IntEnumEntry: Manifest](enum: IntEnum[A]): CustomSerializer[A] = new CustomSerializer[A](_ => (
+    {
+      case JInt(i) if enum.withValueOpt(i.toInt).isDefined => enum.withValue(i.toInt)
+      case JLong(i) if enum.withValueOpt(i.toInt).isDefined => enum.withValue(i.toInt)
```

Comment:
I'm a bit hesitant about doing this, because we could get false matches.

```scala
@ Long.MaxValue 
res0: Long = 9223372036854775807L
@ Long.MaxValue.toInt 
res1: Int = -1
```

Actually, I just realised that `JInt` is a `BigDecimal` (see [here](https://github.com/json4s/json4s/blob/0b28498f065ed8365e4e336a5fec1ec5d4e15649/ast/src/main/scala/org/json4s/JsonAST.scala#L145)), so maybe we should be doing bounds checks for JInt too...